### ISSUE TITLE
No color literals raises warning on variable in map

### ISSUE THREAD

**Issue by brettlangdon** [sasstools/sass-lint#538]
I am seeing an issue where a warning is raised when a color variable is used as a map value. Specifically when the variable shares the same name as a valid CSS color.

To reproduce:

```scss
$black: #000;
$colors: (
  font-color: $black,
);
```

```bash
$ sass-lint --verbose ./test.scss
./test.scss
  3:16  warning  Color literals such as 'black' should not be used as map identifiers  no-color-literals

✖ 1 problem (0 errors, 1 warning)
```

I am running `sass-lint` `1.5.0`.

**Comment by brettlangdon** [comment]
I am working on a pull request for this issue.

The fix I was able to get working was to ensure that the map value's parent was not a `variable`.

**Comment by DanPurdy** [comment]
The reason we decided against was because it actually used to raise the same warnings in scss-lint and we were looking for parity. This fix may have to wait until our 1.6 update is out as we've changed a few of the rules a fair bit.  Thankyou though for taking the time, I'm sure we're far along enough now to diverge from scss-lint

**Comment by brettlangdon** [comment]
More than happy to wait for `1.6`, I have changed my code to use my fork/branch for now anyways, so I can live without this being merged for a little bit.

My only thoughts on the warning are that they are based on the name of the variable, but there isn't a corresponding rule for that variable name not being allowed, the warning would fit better if there was also a rule that disallowed common CSS color names as variables.

**Comment by DanPurdy** [comment]
Yep, I think that makes perfect sense. The whole ethos with `sass-lint` has been for small focused rules so that would suit perfectly.

Thanks again

**Comment by webgefrickel** [comment]
First things first: thanks so much for the long awaited 1.6.0 — almost all issues I had are gone now, well except for this one, I can reproduce this behaviour with 1.6.0. Any ideas on this one?

**Comment by DanPurdy** [comment]
@webgefrickel thanks for the kind words, we're glad 1.6 is sorting a lot of problems out, it's a massive relief for us to get 1.6.0 out with the work we've had to get into it including getting some big issues sorted out in the AST.

As for this issue there's a PR currently open that now 1.6.0 is out we'll be looking to get merged in soon!
#539 

There should be a patch release very very shortly too to fix a few of the extra issues we've come across and more regular updates should now be around as we've got the structural change and AST update out of the way!
